ddtrace/contrib/botocore/patch.py

```diff
@@ -524,27 +524,14 @@ def patched_api_call(original_func, instance, args, kwargs):
                                 and "_datadog" in message["MessageAttributes"]
                                 and "StringValue" in message["MessageAttributes"]["_datadog"]
                             ):
-                                if PROPAGATION_KEY_BASE_64 in json.loads(
-                                    message["MessageAttributes"]["_datadog"]["StringValue"]
-                                ):
-                                    pathway = json.loads(message["MessageAttributes"]["_datadog"]["StringValue"])[
-                                        PROPAGATION_KEY_BASE_64
-                                    ]
-                                    ctx = pin.tracer.data_streams_processor.decode_pathway_b64(pathway)
-                                    ctx.set_checkpoint(["direction:in", "topic:" + queue_name, "type:sqs"])
-
-                                else:
-                                    log.debug(
-                                        "Unable to find data streams context for the message. Data "
-                                        "streams monitoring was not enabled by the application "
-                                        "sending the message at the time of the send."
-                                    )
-                            else:
-                                log.debug(
-                                    "Unable to find trace context for the message. Please ensure that your Datadog "
-                                    "agent is correctly configured on the application sending the message and that "
-                                    "the SQS messages being sent have 9 or less Message Attributes."
+                                pathway = json.loads(message["MessageAttributes"]["_datadog"]["StringValue"]).get(
+                                    PROPAGATION_KEY_BASE_64, None
                                 )
+                            else:
+                                pathway = None
+
+                            ctx = pin.tracer.data_streams_processor.decode_pathway_b64(pathway)
```

Comment:
does this need to be added to the MessageAttributes?

Reply:
`decode_pathway_b64()` expects a string. Passing None should raise an Error.

Reply:
discussed offline, the answer is "no", as we are currently only looking at in/out pairs.